qiskit/dagcircuit/collect_blocks.py

```diff
@@ -86,22 +87,48 @@ def _op_nodes(self):
             return self.dag.get_nodes()
 
     def _direct_preds(self, node):
-        """Returns direct predecessors of a node."""
+        """Returns direct predecessors of a node. This function takes into account the
+        direction of collecting blocks, that is node's predecessors when collecting
+        backwards are the direct successors of a node in the DAG.
+        """
         if not self.is_dag_dependency:
-            return [pred for pred in self.dag.predecessors(node) if isinstance(pred, DAGOpNode)]
+            if self._collect_from_back:
+                return [pred for pred in self.dag.successors(node) if isinstance(pred, DAGOpNode)]
+            else:
+                return [pred for pred in self.dag.predecessors(node) if isinstance(pred, DAGOpNode)]
         else:
-            return [
-                self.dag.get_node(pred_id) for pred_id in self.dag.direct_predecessors(node.node_id)
-            ]
+            if self._collect_from_back:
+                return [
+                    self.dag.get_node(pred_id)
+                    for pred_id in self.dag.direct_successors(node.node_id)
```

Comment:
It's unfortunate that there is a difference like this between `DAGCircuit` and `DAGDependency`.

Reply:
Agreed, there are minor differences in how the methods are called (e.g. `predecessors` vs. `direct_predecessors`), what their arguments are (`DAGNode`/`DAGOpNode` vs. `unsigned` representing node's id) and the type of the underlying node (`DAGNode` vs. `DAGDepNode`). It would indeed be nice to have a common interface for `DAGCircuit` and `DAGDependency`.